### ISSUE TITLE
Do not assume path of local config file

### ISSUE THREAD

**Issue by eggyal** [peter-evans/create-pull-request#1488]
`create-pull-request` currently assumes that the local config can be found in `./.git/config`:

https://github.com/peter-evans/create-pull-request/blob/84c5454c6e165b9ff002651085afb89402cd486a/src/git-auth-helper.ts#L18-L22

However this might not always be true.  In my case, I am in a bare repository, so the config file is actually at `./config`.  Perhaps the git-dir should be identified with a call to `git rev-parse --git-dir` or the local config file itself identified with something like `EDITOR=echo git config --local --edit`?

[Failed workflow](https://github.com/eggyal/copse/actions/runs/3907884293/jobs/6677537599)

**Comment by peter-evans** [comment]
Hi @eggyal 

Thank you for raising this issue. It should be fairly straightforward to identify this with a git call, as you suggest.

When you execute `git rev-parse --git-dir` in your bare repository what is returned? Is it `.`?

**Comment by eggyal** [comment]
Hi @peter-evans!  Thanks for being so receptive (and for the fantastic tool!).

> When you execute git rev-parse --git-dir in your bare repository what is returned? Is it `.`?

Yes, that's right:

```zsh
% git init --bare -q
% git rev-parse --git-dir      
.
```

**Comment by eggyal** [comment]
As an aside, I note that (whilst unlikely) it is *possible* that some configurations may have set the `GIT_CONFIG` envvar in order to use something other than `$GIT_DIR/config`.

**Comment by peter-evans** [comment]
> As an aside, I note that (whilst unlikely) it is possible that some configurations may have set the GIT_CONFIG envvar in order to use something other than $GIT_DIR/config.

For now I'll just handle the git dir. If someone has this issue in future I'll deal with it then.

I've made a fix for this issue and it's currently in v5 beta, which I'm planning to release in the near future. Feel free to use the v5 beta version of the action until it releases.
```yml
      - uses: peter-evans/create-pull-request@v5-rc
```

**Comment by peter-evans** [comment]
This is now fixed in `v5.0.0` / `v5`.

If you are using `v5-beta` or `v5-rc`, please update to `v5`. I will remove pre-release versions after 4 weeks. Thank you for testing the pre-release version!
